src/python/grapl_analyzerlib/grapl_analyzerlib/viewable.py

```diff
@@ -0,0 +1,175 @@
+import abc
+from typing import cast, Any, Dict, Generic, List, Optional, Set, TypeVar, Type, Union, Tuple
+from grapl_analyzerlib.extendable import Extendable
+
+MYPY = False
+
+GraphClient = Any  # todo
+
+V = TypeVar("V", bound="Viewable")
+Q = TypeVar("Q", bound="Queryable")
+T = TypeVar("T")
+OneOrMany = Union[List[T], T]
+
+class Viewable(Generic[V, Q], Extendable, abc.ABC):
+    queryable = None
```

Comment:
queryable: "Type[Q]" = None

i think this works, doesn't even have to be optional, but dont quote me